scripts/test_create2.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+"""Test CREATE2 factory deployment."""
+
+from web3 import Web3
+from eth_utils import keccak
+
+# Connect to Geth
+w3 = Web3(Web3.HTTPProvider("http://127.0.0.1:8545"))
+if not w3.is_connected():
+    print("Failed to connect to Geth")
+    exit(1)
+
+test_account = w3.eth.accounts[0]
+print(f"Using test account: {test_account}")
+print(f"Balance: {w3.eth.get_balance(test_account) / 10**18:.4f} ETH")
+
+# Simple CREATE2 factory that returns the deployed address
+factory_bytecode = (
+    # Constructor: return runtime code
+    "60" + "1f"  # PUSH1 0x1F (runtime size = 31)
+    "80"  # DUP1
+    "60" + "0a"  # PUSH1 0x0A (runtime offset)
+    "60" + "00"  # PUSH1 0x00 (memory dest)
+    "39"  # CODECOPY
+    "60" + "00"  # PUSH1 0x00 (return offset)
+    "f3"  # RETURN
+
+    # Runtime: minimal CREATE2
+    "36"  # CALLDATASIZE
+    "60" + "00"  # PUSH1 0x00
+    "60" + "00"  # PUSH1 0x00
+    "37"  # CALLDATACOPY (copy all calldata to memory)
+
+    "60" + "00"  # PUSH1 0x00 (salt - using 0 for simplicity)
+    "36"  # CALLDATASIZE (size of init code)
+    "60" + "00"  # PUSH1 0x00 (offset in memory)
+    "60" + "00"  # PUSH1 0x00 (value)
+    "f5"  # CREATE2
+
+    "60" + "00"  # PUSH1 0x00
+    "52"  # MSTORE (store address at 0)
+    "60" + "20"  # PUSH1 0x20
+    "60" + "00"  # PUSH1 0x00
```

Comment:
PUSH1 0 could also be written as PUSH0 (`0x5f`) - this is cheaper but not supported on chains which do not have PUSH0. See also https://eips.ethereum.org/EIPS/eip-7997